core/providers/java/java.go

```diff
@@ -0,0 +1,87 @@
+package java
+
+import (
+	"fmt"
+
+	"github.com/railwayapp/railpack/core/generate"
+	"github.com/railwayapp/railpack/core/plan"
+)
+
+type JavaProvider struct{}
+
+func (p *JavaProvider) Name() string {
+	return "java"
+}
+
+func (p *JavaProvider) Detect(ctx *generate.GenerateContext) (bool, error) {
+	return ctx.App.HasMatch("pom.{xml,atom,clj,groovy,rb,scala,yaml,yml}") || ctx.App.HasMatch("gradlew"), nil
+}
+
+func (p *JavaProvider) Initialize(ctx *generate.GenerateContext) error {
+	return nil
+}
+
+func (p *JavaProvider) StartCommandHelp() string {
+	return ""
+}
+
+func (p *JavaProvider) Plan(ctx *generate.GenerateContext) error {
+
+	build := ctx.NewCommandStep("build")
+	build.AddCommand(plan.NewCopyCommand("."))
+	build.Inputs = []plan.Input{plan.NewStepInput(ctx.GetMiseStepBuilder().Name())}
+
+	if p.usesGradle(ctx) {
+		p.setGradleVersion(ctx)
+		p.setJDKVersion(ctx, nil)
+
+		if ctx.App.HasMatch("gradlew") && !ctx.App.IsFileExecutable("gradlew") {
+			build.AddCommand(plan.NewExecCommand("chmod +x gradlew"))
+		}
+
+		build.AddCommand(plan.NewExecCommand("./gradlew clean build -x check -x test"))
```

Comment:
Is `test` needed when building an image? Builds will be faster if it is removed and railpack is not intended to be a CI platform.

Reply:
`-x check -x test` excludes `check` and `test`.